### PR TITLE
[BugFix] Fix NPE due to fail to get rollup function name

### PR DESCRIPTION
**Pull request by kaijianding**
## What I'm doing:
`new FunctionName(name)` wil verify name is not null, then NPE will be thrown when rollupFunctionName is null
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewPushDownRewriter.java [hunk 1]
```diff
@@ -396,6 +396,10 @@ public AggRewriteInfo visitLogicalAggregate(OptExpression optExpression, AggRewr
                     List<ScalarOperator> newArgs = aggCall.getChildren();
                     newArgs.set(0, newArg0);
                     String rollupFuncName = getRollupFunctionName(aggCall, false);
+                    if (rollupFuncName == null) {
```
Comment (alvin-celerdata) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewPushDownRewriter.java:399]: When will rollupFuncName be null?

Comment (kaijianding) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewPushDownRewriter.java:399]: It happens when rollup+count(distinct)

```sql
CREATE TABLE test_pt (
  id int(11) NULL COMMENT "id",
  pt date NOT NULL COMMENT "",
  gmv int(11) NULL COMMENT "gmv"
) ENGINE=OLAP
DUPLICATE KEY(id)
COMMENT "OLAP"
PARTITION BY date_trunc('day', pt)
DISTRIBUTED BY HASH(pt)
PROPERTIES (
"replication_num" = "1"
);


CREATE MATERIALIZED VIEW test_pt_mv3
PARTITION BY (pt)
DISTRIBUTED BY hash(id) buckets 1
REFRESH ASYNC START("2024-03-08 03:00:00") EVERY(INTERVAL 1 DAY)
PROPERTIES (
"replication_num" = "1"
)
AS SELECT pt, id, gmv,count(1) AS pv
FROM test.test_pt
GROUP BY pt, id,gmv;


explain
select 
    count(distinct gmv) from test_pt
    where pt='2023-03-07' 
    and id IN (
        SELECT
            id
        FROM
            test_alias
    )
    group by
    id;

```

`count(distinct gmv)` will trigger this NPE when mv's group by keys contains `gmv` and the query's group by key is only `id` which is `rollup`

Comment (alvin-celerdata) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewPushDownRewriter.java:399]: Thank you for the explanation, better to add comment in the code base

Comment (kaijianding) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewPushDownRewriter.java:400]: Added.

**Review by LiShuMing — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**